### PR TITLE
Setup Script

### PR DESCRIPTION
**Pull request by jordygarcias**
What does this PR do?
It implements a new script to set up the project, to avoid developers to replace by themselves the application name.

Any background context you want to provide?
This setup script includes running npm install command on the API and UI packages.

To run this new script, the developers need to run npm run setup --appName= --shortName= if shortName is not provided, the script will use appName instead.

### CODE REVIEW

File: serverless.yaml [hunk 1]
```diff
@@ -165,12 +165,13 @@ functions:
...
         Resource:
           - !GetAtt UserTable.Arn
-  
+
   cognitoAutoConfirm:
     handler: packages/api/functions/cognito/auto-confirm-user.handler
 
   cognitoPostAuthN:
     handler: packages/api/functions/cognito/post-authentication.handler
+    iamRoleStatementsName: serverless-cognito-post-auth
```
Comment (brettstack) [inline serverless.yaml:174]: Need to remove this. I know you added it due to the name length limit, but we can't have hardcoded names as it breaks deploying multiple versions into a single account. Maybe find the limit and add a check in the setup script to print an error if the provided app name exceeds the length? Alternatively, we can reduce the resource names for `cognitoAutoConfirm` and `cognitoPostAuthN`.

File: scripts/installDependencies.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      console.error('An error has ocurred while trying to install npm dependencies in API Package', err)
+    } else {
+      console.info('NPM Dependencies installed in API Package', data)
+    }
+  }))
+}
+
+const installDependencies = async () => {
+  await installUIDependencies()
+  await installAPIDependencies()
```
[the remark this answers is not in the document]
Comment (jordygarcias) [reply scripts/installDependencies.js]: Yeah, done!

File: scripts/installDependencies.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+const npm = require('npm')
+const constants = require('./constants')
+
+const installUIDependencies = () => {
+  npm.load(() => npm.commands.install(constants.UIPath, [], (err, data) => {
```
[the remark this answers is not in the document]
Comment (jordygarcias) [reply scripts/installDependencies.js]: There's not a Promise variant with npm package. We should have to use another package.

And about the logs, they were removed of the process, it actually logs the install process, we don't need an extra message.

File: scripts/replaceFlags.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+const replace = require('replace-in-file')
+const constants = require('./constants')
+
+const getUndescoredString = (string) => string.split(' ').join('_')
+const getLowerCaseString = (string) => string.toLowerCase()
+const getApplicationName = (string) => getUndescoredString(getLowerCaseString(string))
```
[the remark this answers is not in the document]
Comment (jordygarcias) [reply scripts/replaceFlags.js]: Yeah, done!

File: serverless.yaml (hunk 1 above)
Comment (jordygarcias) [reply serverless.yaml]: Ok, understood 😄

File: scripts/constants.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+const flags = [/myapp/g, /MyApp/g]
+const longNameFlags = [/My App/g]
+
+/* DIR PATHS */
+const UIPath = 'packages/ui/'
+const APIPath = 'packages/api/'
+
+const packageJson = 'package.json'
+const yamlFile = 'serverless.yaml'
```
Comment (NellyGzzA) [inline scripts/constants.js:9]: (nit) I'm not 100% sure about `yamlFile` because it isn't descriptive as our other constants

**Pull request merged**